File: aplicacion/lib/models/cliente.dart
/// Cliente de la empresa. El cliente "S/N" (sin nombre, [esDefault]) es el
/// que usa la venta cuando no se elige otro; no se edita ni se borra.
class Cliente {
  final int id;
  final String nombre;
  final String? nit;
  final String? telefono;
  final String? correo;
  final String? direccion;
  final bool esDefault;

  const Cliente({
    required this.id,
    required this.nombre,
    this.nit,
    this.telefono,
    this.correo,
    this.direccion,
    this.esDefault = false,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) => Cliente(
        id: json['id'] as int,
        nombre: json['nombre'] as String,
        nit: json['nit'] as String?,
        telefono: json['telefono'] as String?,
        correo: json['correo'] as String?,
        direccion: json['direccion'] as String?,
        esDefault: json['es_default'] == true || json['es_default'] == 1,
      );

  /// Nombre para mostrar: el S/N se explica solo.
  String get etiqueta => esDefault ? 'S/N (sin nombre)' : nombre;
}

File: back/app/Models/Cliente.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;
use OwenIt\Auditing\Auditable;
use OwenIt\Auditing\Contracts\Auditable as AuditableContract;

#[Fillable(['empresa_id', 'nombre', 'nit', 'telefono', 'correo', 'direccion', 'es_default'])]
class Cliente extends Model implements AuditableContract
{
    use Auditable, SoftDeletes;

    protected $table = 'clientes';

    protected function casts(): array
    {
        return [
            'es_default' => 'boolean',
        ];
    }

    public function empresa(): BelongsTo
    {
        return $this->belongsTo(Empresa::class);
    }

    /**
     * El cliente por defecto de la empresa ("S/N", sin nombre): se usa en la
     * venta cuando no se elige otro. Si la empresa aún no lo tiene (empresas
     * creadas antes de este módulo), se crea aquí.
     */
    public static function porDefecto(Empresa $empresa): self
    {
        return self::firstOrCreate(
            ['empresa_id' => $empresa->id, 'es_default' => true],
            ['nombre' => 'S/N'],
        );
    }
}
